### PR TITLE
fix(OverlayBadge): [SPC-1728] fix number alignment on small variant

### PR DESCRIPTION
**Pull request by hmschreiner**
[SPC-1728]

### Description

Fix the number alignment for the `OverlayBadge` component on the `small` variant.

### How to test

- Check the `small` variant to the `OverlayBadge` component
- Change the page zoom level to 125%
- The number alignment should keep on center

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Screen Shot 2022-03-24 at 16 23 24 | Screen Shot 2022-03-24 at 16 23 47 |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[SPC-1728]: https://toptal-core.atlassian.net/browse/SPC-1728?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hmschreiner** [comment]
@toptal-bot ping reviewers

**Review by bogdanalexe90 — APPROVED** [review]

File: .changeset/little-squids-jog.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': patch
+---
+
+fix(OverlayBadge): fix number alignment on small variant
```
Comment (denieler) [inline .changeset/little-squids-jog.md:5]: please follow changeset guidelines 🙇‍♂️ https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md

File: .changeset/little-squids-jog.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@toptal/picasso': patch
+---
+
```
Comment (denieler) [inline .changeset/little-squids-jog.md:4]: here should be one more `---`

**Review by OndrejTuma — APPROVED** [review]
Looks good, thank you 👍

File: .changeset/little-squids-jog.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+### OverlayBadge
+
+- Fix number alignment on small variant
```
Comment (OndrejTuma) [inline .changeset/little-squids-jog.md:7]: nit:
```suggestion
- Fix line-height on small variant
```

File: .changeset/little-squids-jog.md (hunk 2 above)
Comment (hmschreiner) [reply .changeset/little-squids-jog.md:4]: Done

**Review by denieler — APPROVED** [review]

**Pull request merged**